packages/screens/Mini/Wallet/TokenScreen.tsx

```diff
@@ -33,15 +34,16 @@ import {
 import { layout } from "@/utils/style/layout";
 
 const TokenScreen: ScreenFC<"MiniWallets"> = ({ navigation }) => {
-  const wallets = useSelector(selectAllWallets);
+  const wallets = useSelector(selectAllWallets) as StoreWallet[];
   const dispatch = useAppDispatch();
-  if (wallets.length === 0) {
-    navigation.navigate("NativeWallet");
-  }
 
   const selectedWallet = useSelectedNativeWallet();
   if (!selectedWallet) {
-    dispatch(setSelectedNativeWalletIndex(wallets[0].index));
+    if (wallets.length !== 0) {
+      dispatch(setSelectedNativeWalletIndex(wallets[0].index));
+    } else {
+      navigation.navigate("NativeWallet");
+    }
   }
```

Comment:
calling dispatch and navigate in render function is very bad and can cause all sorts of crashes and weird behaviors, it should be in an effect or action callback

Reply:
a component function should be a "pure" function that transforms it's props into jsx, side-effects should only happen in effects (useEffect) and callbacks (onPress, onTextChange, etc..)

Reply:
see https://react.dev/learn/render-and-commit

Reply:
is this related https://react.dev/learn/you-might-not-need-an-effect ?

Reply:
well, this is the counterpart, people often overuse effects and memos when it's not needed, both are interesting but it's not the related to the problems on these lines